osquery/database/database.cpp

```diff
@@ -674,14 +674,15 @@ Status scanDatabaseKeys(const std::string& domain,
     }
     return status;
   } else if (!DatabasePlugin::kDBInitialized) {
-    throw std::runtime_error("Cannot get database values");
+    throw std::runtime_error("Cannot scan database keys");
   } else {
     auto plugin = getDatabasePlugin();
     return plugin->scan(domain, keys, prefix, max);
   }
 }
 
 void resetDatabase() {
+  WriteLock lock(kDatabaseReset);
```

Comment:
I think this will cause a out-of-order lock if locked before the calls to `Registry::getActive`.